[Delivers-#172534740] Lobby

AdriAlvarez00 opened the pull request:
### LOBBY
No esta integrado, ya que no hay estructura en la que integrarlo aun, pero el lobby es capaz de

- Detectar cuando se unen o abandonan los jugadores, y los guarda. Los mandos se unen con la A/X, el teclado se une con W o I y el raton se conecta con click izquierdo y desconecta con click derecho.

- Si un jugador conectado pulsa su tecla de avanzar (ver ampliacion InputBinder) pasa a estar listo, y con la tecla de back puede ir hacia atras o desconectarse

### AMPLIACION INPUTBINDER
Se han añadido metodos de movimiento en menus en la clase InputBinder, en el futuro habra que dejar de pedir informacion en los menus a inputHandler y pedirla a partir de binder

eljorgemo committed:
Clase LobbyState

eljorgemo committed:
filtros lol

AdriAlvarez00 committed:
Permite unirse y abandonar a mandos

AdriAlvarez00 committed:
Permite unirse y abandonar a los pureKeyboards

eljorgemo committed:
algunos fallitos y comienzo del render

eljorgemo committed:
Update LobbyState.cpp

eljorgemo committed:
recogida basura, refactor, y se renderiza en pantalla

eljorgemo committed:
Mucho refactor y arreglo de fallos

AdriAlvarez00 committed:
Arreglada excepcion e iconitos

eljorgemo committed:
más refactor

eljorgemo committed:
botones por variable

AdriAlvarez00 committed:
Los jugadores pueden indicar si estan listos

eljorgemo committed:
Skins y ratón

AdriAlvarez00 committed:
Merge branch 'master' into 172534740-lobby-multijugador

AdriAlvarez00 committed:
Fix de casos especiales en los que no te podias desconectar

WillMDev commented on src/SourceFiles/InputBinder.h:
```diff
@@ -86,6 +91,10 @@ class InputBinder
 	virtual bool holdImpulse() = 0;
 	virtual bool releaseImpulse() = 0;
 	virtual bool pressAttack() = 0;
+	virtual void setPlayerCol(Collider* col) {};
```
No debería ser
virtual void setPlayerCol(Collider* col) = 0;
si InputBinder es abstracta pura?

AdriAlvarez00 replied:
Es intencionado que eche a PureKeyboard2 cuando conectas teclado porque sería bastante incómodo.
SetPlayerCol no es =0 porque por defecto no hace nada, por ahora solo tiene utilizad en el MouseKeyboardBinder ya que hay que sacar la dirección entre su centro y el sitio al que hace click
La excepción es, efectivamente, una exccepción que se nos habia pasado ytenemos que arreglar

WillMDev reviewed (changes requested):
-Entiendo que al seleccionar mouse el jugador 1, quitar al jugador 2 si tiene teclado es intencionado. Entonces este estado seria no deseado no?
![unknown](https://user-images.githubusercontent.com/43720307/80893307-7f4c9400-8cd1-11ea-81db-9646067eda64.png)

Para reproducirlo: **W -> I -> Q -> clickIzq**

-También me salta la ventana de error Debug Assertion Failed
Para reproducirlo: **W -> clickIzq -> Q -> clickDerch**

AdriAlvarez00 committed:
Solucionada excepción

Resulta que cuando se cambiaba de teclado y raton no se cambiaba bien el identificador

EstebenR reviewed (commented):
me gustaria que aunque aun no este implementado el wrapper, cuando todos hayan aceptado se pueda ir al menú para facilitar la prueba de otras ramas

EstebenR reviewed (changes requested):
me gustaria que aunque aun no este implementado el wrapper, cuando todos hayan aceptado se pueda ir al menú para facilitar la prueba de otras ramas

AdriAlvarez00 committed:
Ea, ya se puede ir al menu
